lxd/archive/archive.go

```diff
@@ -0,0 +1,254 @@
+package archive
+
+import (
+	"archive/tar"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+
+	"golang.org/x/sys/unix"
+
+	"github.com/lxc/lxd/lxd/apparmor"
+	"github.com/lxc/lxd/lxd/sys"
+	"github.com/lxc/lxd/shared"
+	"github.com/lxc/lxd/shared/ioprogress"
+	"github.com/lxc/lxd/shared/logger"
+	"github.com/lxc/lxd/shared/subprocess"
+)
+
+// ExtractWithFds runs extractor process under specifc AppArmor profile.
+// The allowedCmds argument specify commands which are allowed to run by apparmor.
+// The cmd argument is automatically added to allowedCmds slice.
+func ExtractWithFds(cmd string, args []string, allowedCmds []string, stdin io.ReadCloser, sysOS *sys.OS, output *os.File) error {
+	outputPath := output.Name()
+
+	allowedCmds = append(allowedCmds, cmd)
+	allowedCmdsPath := []string{}
+	for _, c := range allowedCmds {
+		cmdPath, err := exec.LookPath(c)
+		if err != nil {
+			return fmt.Errorf("Failed to start unpack: Failed to find executable: %w", err)
```

Comment:
Should we include the `c` var in this error or if its missing does Go already include it in the `err` var?

Reply:
Just a nit, should the error prefix be `Failed to start extract` (as the function is named "Extract" rather than "Unpack")?

Reply:
Error message looks like:
`Failed to find executable: exec: "test1": executable file not found in $PATH`
so I think that there is no need to include `c` var.